iceoryx_posh/include/iceoryx_posh/roudi/iceoryx_roudi_components.hpp

```diff
@@ -17,30 +17,45 @@
 #define IOX_POSH_ROUDI_ICEORYX_ROUDI_COMPONENTS_HPP
 
 #include "iceoryx_posh/internal/roudi/port_manager.hpp"
-#include "iceoryx_posh/internal/roudi/roudi_lock.hpp"
 #include "iceoryx_posh/roudi/memory/iceoryx_roudi_memory_manager.hpp"
 #include "iceoryx_utils/cxx/expected.hpp"
 #include "iceoryx_utils/cxx/generic_raii.hpp"
+#include "iceoryx_utils/posix_wrapper/file_lock.hpp"
+
 
 namespace iox
 {
 namespace roudi
 {
+constexpr char ROUDI_LOCK_NAME[] = "unique-roudi";
 struct IceOryxRouDiComponents
 {
   public:
     IceOryxRouDiComponents(const RouDiConfig_t& roudiConfig) noexcept;
 
     virtual ~IceOryxRouDiComponents() = default;
 
-    /// @brief Locks the socket for preventing multiple start of RouDi
-    RouDiLock m_roudilock;
+    posix::FileLock fileLock = std::move(posix::FileLock::create(ROUDI_LOCK_NAME)
+                                             .or_else([](auto& error) {
+                                                 if (error == posix::FileLockError::LOCKED_BY_OTHER_PROCESS)
+                                                 {
+                                                     LogFatal() << "Could not acquire lock, is RouDi still running?";
+                                                     exit(EXIT_FAILURE);
```

Comment:
We should use `std::terminate` instead of `exit`.

Reply:
Hmm, thinking about it again. We shouldn't be doing either of those but call the errorHandler instead, ack?

Reply:
Added two new errors.